Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+package com.earth2me.essentials.services;
+
+import com.earth2me.essentials.Essentials;
+import com.earth2me.essentials.User;
+import com.earth2me.essentials.api.NoLoanPermittedException;
+import com.earth2me.essentials.api.UserDoesNotExistException;
+import net.ess3.api.Economy;
+import net.tnemc.core.Reserve;
+import net.tnemc.core.economy.EconomyAPI;
+import net.tnemc.core.economy.currency.Currency;
+import org.bukkit.World;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+/**
+ * Reserve economy provider implementation for EssentialsX.
+ * @author creatorfromhell
+ */
+public class ReserveEssentials implements EconomyAPI {
+
+    private final Essentials ess;
+
+    private ReserveEssentials(Essentials ess) {
+        this.ess = ess;
+    }
+
+    /**
+     * Register the provider if no other economy is available.
+     *
+     * @param ess The Essentials plugin instance
+     */
+    public static void register(Essentials ess) {
+        Reserve reserve = Reserve.instance();
+
+        // Check to see if there is an economy provider already registered.
+        // If so, the existing provider should take priority
+        if (!reserve.economyProvided()) {
+            Reserve.instance().registerProvider(new ReserveEssentials(ess));
+        }
+    }
+
+    /**
+     * @return The name of the Economy implementation.
+     */
+    @Override
+    public String name() {
+        return "Essentials Economy";
+    }
+
+    /**
+     * @return The version of Reserve the Economy implementation supports.
+     */
+    @Override
+    public String version() {
+        return "0.1.2.0";
+    }
+
+    //This is our method to convert UUID -> username for use with Essentials' create account methods.
+    private String getName(UUID identifier) {
+        final User user = ess.getUser(identifier);
+        return user == null ? identifier.toString() : user.getName();
+    }
+
+    /**
+     * @return Whether or not this implementation is enabled.
+     */
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    /**
+     * @return Whether or not this implementation should have a default Vault implementation.
+     */
+    @Override
+    public boolean vault() {
+        return false;
+    }
+
+    /**
+     * Used to get the plural name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural() {
+        return "Dollars";
```

Comment:
This feels like something that Ess is localizing / should be localizing.

Reply:
Essentials doesn't have a name for its currency from what I've seen. It simply says "money."

Reply:
So this should still be localized in some way since 'Dollars' is a region specific term for currency.

Reply:
I added two new strings to messages.properties. If this is the route that should be gone down.